### DISCUSSION TITLE
What's preventing my public key from being reachable? Step-by-step config

### DISCUSSION THREAD

**Discussion by kmil4** [llamafilm/tesla-http-proxy-addon#65] [Q&A]
I've been hitting a wall with this add-on so I'm transcribing my approach step-by-step in hopes that someone can tell me where I've gone wrong. I couldn't send commands with the owner API anymore so I'm trying to migrate. Pardon unnecessary redactions, this isn't my expertise so I don't know what's risky to share.

First, my environment:

- RPi4+ running HAOS
- OS and HA on the latest stable versions
- Remote access via Nabu Casa, but when I'm local I simply hit `192.168.0.2:8123`
- External SSD for storage on the Pi and using PoE for power
- Wiped my file system clean of all traces of previous telsa configs

## Step-by-Step Configuration
Following the docs to the best of my ability.

### 1. Registered domain on DuckDNS:
`theredrocket.duckdns.org`

I've registered the domain, I've got an IP address and token.

### 2. Install and configure DuckDNS Add-On

Here's the config. The only "novel" thing is my use of aliases, this seemed to be the only way I could ge LetsEncrypt to get me a cert for `tesla.theredrocket.duckdns.org` so that's why it's there:
```
domains:
  - theredrocket.duckdns.org
token: xxxxxxx-xxxxx-xxxx-xxxx-xxxxxxxxx
aliases:
  - domain: theredrocket.duckdns.org
    alias: tesla.theredrocket.duckdns.org
  - domain: theredrocket.duckdns.org
    alias: "*.theredrocket.duckdns.org"
lets_encrypt:
  accept_terms: true
  algo: secp384r1
  certfile: fullchain.pem
  keyfile: privkey.pem
seconds: 300
```

Start add-on and see the log output. I can now see `fullchain.pem` and `privkey.pem` under my `/ssl` directory. FYI this was the original log before adding those aliases above (but that was successful too).

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
# INFO: Using main config file /data/workdir/config
+ Account already registered!
[09:35:18] INFO: OK
// IP address was here

NOCHANGE
[09:35:18] INFO: Renew certificate for domains: theredrocket.duckdns.org and aliases: 
# INFO: Using main config file /data/workdir/config
Processing theredrocket.duckdns.org
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting new certificate order from CA...
 + Received 1 authorizations URLs from the CA
 + Handling authorization for theredrocket.duckdns.org
 + 1 pending challenge(s)
 + Deploying challenge tokens...
OK + Responding to challenge for theredrocket.duckdns.org authorization...
 + Challenge is valid!
 + Cleaning challenge tokens...
OK + Requesting certificate...
 + Checking certificate...
 + Done!
 + Creating fullchain.pem...
 + Done!
```

### 3. Install and configure LetsEncrypt Add-On

Config:
```
domains:
  - tesla.theredrocket.duckdns.org
email: (private)@gmail.com
keyfile: privkey.pem
certfile: fullchain.pem
challenge: dns
dns:
  provider: dns-duckdns
  duckdns_token: xxxxxxx-xxxxx-xxxx-xxxx-xxxxxxxxx
```

Log output after starting LetsEncrypt:
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/file-structure.sh
cont-init: info: /etc/cont-init.d/file-structure.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun lets-encrypt (no readiness notification)
s6-rc: info: service legacy-services successfully started
[10:38:11] INFO: Selected DNS Provider: dns-duckdns
[10:38:11] INFO: Use propagation seconds: 60
[10:38:11] INFO: Detecting existing certificate type for tesla.theredrocket.duckdns.org
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for tesla.theredrocket.duckdns.org
Waiting 60 seconds for DNS changes to propagate

Successfully received certificate.
Certificate is saved at: /data/letsencrypt/live/tesla.theredrocket.duckdns.org/fullchain.pem
Key is saved at:         /data/letsencrypt/live/tesla.theredrocket.duckdns.org/privkey.pem
This certificate expires on 2024-06-25.
These files will be updated when the certificate renews.
NEXT STEPS:
- The certificate will need to be renewed before it expires. Certbot can automatically renew the certificate in the background, but you may need to take steps to enable that functionality. See https://certbot.org/renewal-setup for instructions.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
If you like Certbot, please consider supporting our work by:
 * Donating to ISRG / Let's Encrypt:   https://letsencrypt.org/donate
 * Donating to EFF:                    https://eff.org/donate-le
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

### 4. Install and configure NGINX Home Assistant SSL Proxy Add-On

Before configuring the add-on, I'm adding this to my `configuration.yaml` and restarting home assistant
```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24
```

After restart, I'm configuring the add-on as follows:
```
domain: tesla.theredrocket.duckdns.org
hsts: max-age=31536000; includeSubDomains
certfile: fullchain.pem
keyfile: privkey.pem
cloudflare: false
customize:
  active: true
  default: nginx_proxy_default*.conf
  servers: nginx_proxy/*.conf
```

Starting the add-on, here are the logs:
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service nginx: starting
s6-rc: info: service nginx successfully started
s6-rc: info: service crond: starting
s6-rc: info: service crond successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[09:47:14] INFO: Generating dhparams (this will take some time)...
Generating DSA parameters, 4096 bit long prime
.....+..+......+.......+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*
..+...+..+................+..............+....................+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*
[09:47:31] INFO: Running nginx...
```

### 4. Install and configure Tesla HTTP Proxy

Here's the config, leaving client ID and secret blank for now per the documentation:
```
client_id: ""
client_secret: ""
domain: tesla.theredrocket.duckdns.org
debug: true
regenerate_auth: true
region: North America, Asia-Pacific
```

Now I start the add-on, and get this log:

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service webui: starting
s6-rc: info: service webui successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[11:02:15] webui:INFO: Starting Flask server for Web UI...
[11:02:15] werkzeug:INFO: WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8099
 * Running on http://172.30.33.9:8099
[11:02:15] werkzeug:INFO: Press CTRL+C to quit
[11:02:16] INFO: Setting up GnuPG and password-store
gpg: keybox '/data/gnugpg/pubring.kbx' created
gpg: /data/gnugpg/trustdb.gpg: trustdb created
gpg: directory '/data/gnugpg/openpgp-revocs.d' created
gpg: revocation certificate stored as '/data/gnugpg/openpgp-revocs.d/(blinded).rev'
gpg: checking the trustdb
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: next trustdb check due at 2027-03-27
/data/gnugpg/pubring.kbx
------------------------
pub   (blinded for security) [SC] [expires: 2027-03-27]
      (blinded for security)
uid           [ultimate] myself
sub   (blinded for security) 2024-03-27 [E]
created directory: '/data/password-store/'
Password store initialized for myself
[11:02:16] INFO: Current Nginx configuration
{"domain":"theredrocket.duckdns.org","hsts":"max-age=31536000; includeSubDomains","certfile":"fullchain.pem","keyfile":"privkey.pem","cloudflare":false,"customize":{"active":true,"default":"nginx_proxy_default*.conf","servers":"nginx_proxy/*.conf"}}
[11:02:16] INFO: Adding custom config to /share/nginx_proxy/nginx_tesla.conf
    server {
        server_name tesla.theredrocket.duckdns.org;

        ssl_certificate /ssl/fullchain.pem;
        ssl_certificate_key /ssl/privkey.pem;

        # dhparams file
        ssl_dhparam /data/dhparams.pem;

        listen 443 ssl http2;
        add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;

        # static public key for Tesla
        location /.well-known/appspecific/com.tesla.3p.public-key.pem {
            root /share/tesla;
            try_files /com.tesla.3p.public-key.pem =404;
        }

        location = /favicon.ico {
            log_not_found off;
        }

        location = /robots.txt {
            log_not_found off;
        }
    }
[11:02:16] INFO: Generating self-signed SSL certificate
-----
[11:02:16] INFO: Generating keypair
-----BEGIN PUBLIC KEY-----
(REDACTED FOR SECURITY, BUT THIS WORKED)==
-----END PUBLIC KEY-----
[11:02:17] INFO: Testing public key...
[11:02:22] FATAL: Fix public key before proceeding.
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service webui: stopping
s6-rc: info: service webui successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

Then, I called the supervisor service to STOP the NGINX add-on, and restarted it. 

### The Result...

I can see `/nginx_proxy` and `/tesla` in my `/share` directory (using Samba add-on for Windows). I see files appearing in each including public key. 

**I still cannot reach the public key** from `https://tesla.theredrocket.duckdns.org/.well-known/appspecific/com.tesla.3p.public-key.pem`. I hit a 404 page every time.

I've looked up `theredrocket.duckdns.org` and `tesla.theredrocket.duckdns.org` using DNS checkers and both are returning results.

Where did I go wrong?

**Comment by llamafilm** [comment]
It will be easier to help if you show the correct domain name. That piece of information is not sensitive; the public key is meant to public.

**Reply by kmil4** [reply]
@llamafilm OK sounds good. I've removed that redaction so any instance of `blueberry.duckdns.org` is now `theredrocket.duckdns.org` which is real.

**Reply by llamafilm** [reply]
I can see your DNS resolves at least, so that's good.  Assuming 73.42.203.35 is your correct IP address, it sounds like nginx is not configured right, maybe it's not seeing the `/share/nginx_proxy/nginx_tesla.conf` file.  What do you see in the nginx log?

**Reply by kmil4** [reply]
The NGINX log is in the original post.

Appreciate the help but JFYI I've decided to switch to Tessie instead. I've spent too much time in this so decided to cut losses

**Comment by gimmel** [comment]
I have the same problems. I don’t even know if I really need a second domain for the Tesla stuff. The tutorial for the Tesla HTTP Proxy recommends „NGINX Home Assistant SSL proxy“ which can only use one domain. I used the community add-on „Nginx Proxy Manager“ before, which can use multiple domains, but it didn’t work with Tesla HTTP Proxy tutorial.

**Comment by Briochorama** [comment]
> I've been hitting a wall with this add-on so I'm transcribing my approach step-by-step in hopes that someone can tell me where I've gone wrong. I couldn't send commands with the owner API anymore so I'm trying to migrate. Pardon unnecessary redactions, this isn't my expertise so I don't know what's risky to share.
> 
> First, my environment:
> [...]
> Where did I go wrong?

I'm in a similar situation : I've setup duckdns + nginx and can access my HA instance from duckdns url. However, 404 error when trying to reach ` https://tesla.mydomain.duckdns.org/.well-known/appspecific/com.tesla.3p.public-key.pem.`.

Two questions :

- Do you have a nabu casa account ? I have, and I'm wondering if this could cause issues.
- Why is the lets encrypt addon used in your config ?

**Reply by llamafilm** [reply]
What do you see in Nginx logs?

**Reply by Briochorama** [reply]
Nothing much, here's the full log :

```
NGINX Home Assistant SSL proxy
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service nginx: starting
s6-rc: info: service nginx successfully started
s6-rc: info: service crond: starting
s6-rc: info: service crond successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[19:30:48] INFO: Running nginx...
nginx: [warn] conflicting server name "mydomain.duckdns.org" on 0.0.0.0:443, ignored
```

This is the same after each reboot of HA and Nginx

**Reply by llamafilm** [reply]
Usually a 404 error would be displayed in the log. So this suggests that 404 is coming from somewhere other than Nginx. Are you sure your public IP is correct and your router is NAT port forwarding is working?

**Reply by Briochorama** [reply]
> Usually a 404 error would be displayed in the log. So this suggests that 404 is coming from somewhere other than Nginx. Are you sure your public IP is correct and your router is NAT port forwarding is working?

I'd say yes to both, here's why. 

IP - I have a same values between :
- A sensor in HA that displays my current IP from my unifi router;
- The duckdns website and duckdns addons logs.
- [What is my ip.com](https://www.whatismyip.com/)

Port forwarding is setup in the router as follow (internal IP is the one of my HA instance, a proxmox VM):
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/12542442/7fb71186-7b8c-49ff-a0a6-0d7410f9f7ab)

Thing is, when I type the url of my duckdns domain I land on HA login page :
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/12542442/2f086bed-e5a5-48c5-a4bd-80239a68f4d5)

So i'd say, at the very least, there's something working there because if I stop either the duckdns addon or nginx, I can't reach this page (obviously). There's no mention of anything in the log each time I naviguate to this url.

**important edit:** I also have a nabu casa subscription, meaning I can access my HA instance via https://mynabucasaurl.ui.nabu.casa/. Could there be a conflict here ?

**Reply by llamafilm** [reply]
Ok, so it sounds like Nginx is receiving the request. I think you need to fix this:

nginx: [warn] conflicting server name "mydomain.duckdns.org" on 0.0.0.0:443, ignored

If you share the full unredacted contents of all Nginx config files, it would be easier to help you. I really don’t understand why people are shy about sharing a domain name.

**Reply by llamafilm** [reply]
@SmarterHomeLife please share the base config of Nginx addon as well, from Settings > Addons > Nginx > Configuration > Edit in YAML.  I suspect you are trying to use the same `server_name` for both, which won't work.

**Reply by llamafilm** [reply]
Not quite.  What I meant is that you no longer need a 2nd domain name.  So if you don't want Home Assistant to be publicly accessible on its own domain, then make up a fake name for the "primary" domain, i.e. change the first line to this:
```yaml
domain: core-nginx-proxy
```

**Reply by llamafilm** [reply]
yes

**Reply by Briochorama** [reply]
changing the domain in nginx to `domain: core-nginx-proxy` fixed both issues. Key is reachable and no more conflict error. I'm moving forward with the rest of the procedure !

**Comment by kmil4** [comment]
I'll leave this open since others are troubleshooting, but I'm no longer in need of a fix.

I gave up and went with Tessie. The integration is fantastic and took all of 3 minutes to start sending commands to my car again.

To others banging their heads against a wall over this, if you can afford $5/month for Tessie, highly recommended too save yourself the trouble.

**Comment by tannerwastaken** [comment]


**Reply by llamafilm** [reply]
That's odd.  Your key looks okay to me.  It might be a local DNS problem, related to hairpin NAT loopback.  Does that URL work from inside the addon container?  If you install the advanced SSH addon, then you can SSH into the Home Assistant host, and from there run `docker exec addon_c03d64a7_tesla_http_proxy curl -v https://tesla.thecolorsofyou.com/.well-known/appsp
ecific/com.tesla.3p.public-key.pem`.

**Reply by llamafilm** [reply]
HAOS is built on docker. My previous suggestion is meant for HAOS.

**Reply by tannerwastaken** [reply]
Okay, my bad. I'm not the best with this kind of stuff (trying though). This is the response: 
<img width="908" alt="Screenshot 2024-03-30 at 5 10 15 PM" src="https://github.com/llamafilm/tesla-http-proxy-addon/assets/121889392/5026717a-f223-4f0e-8d5a-a88995075454">

**Reply by tannerwastaken** [reply]
Okay, I figured it might be because the add-on isn't running. so I ran it and it just hangs... Currently I have port forwarding setup for 80 / 443 > 8126 (the port I chose for nginx)
<img width="589" alt="Screenshot 2024-03-30 at 5 13 24 PM" src="https://github.com/llamafilm/tesla-http-proxy-addon/assets/121889392/02930189-7032-4096-b226-9387c8c18bd1">

**Reply by llamafilm** [reply]
That screenshot appears to be the regular SSH addon.  For this troubleshooting step, you need to be in the Docker _host_.  You can install the port 22222 configurator from here: https://github.com/adamoutler/HassOSConfigurator.  Then run it and SSH in on port 22222.

**Reply by baylanger** [reply]
Does the T HTTP Proxy addon exit if there’s a failure to check the key? That would explain why the container ain’t running.

If you run the curl command with its arguments directly in a HAOS SSH terminal, does that give an error? Seems it should work from there as well.

**Reply by baylanger** [reply]
What does nslookup or dig returns for the IP address using your Tesla hostname as argument?

If it returns the external IP address, I’m not sure if basic port forwarding on your router will properly forward the traffic coming from your internal network. I’m just not sure. Can a PC connected on your local network access the key using the URL to your key?

**Reply by baylanger** [reply]
53 is the default tcp/udp port for DNS service.

Per the output of the command, can you do a nslookup tesla.thecolorsofyou.com , not thecolorsofyou.com - perhaps both are pointing to the same local IP?!

If they don't have the same IP address, from your PC did you try to access the key using https://tesla.thecolorsofyou.com/... or just https://thecolorsofyou.com ?

Last. the nslookup you ran is within HAOS SSH terminal or your PC? Try in HAOS.

**Reply by chowdarygm** [reply]
Did you find a solution for this?

**Comment by baylanger** [comment]
> Ah yes, the subdomain is pointing to the same local IP as my root domain - is that wrong? How should I configure the dns for the subdomain?
>
Since you're doing port forwarding on your router, it's likely that both should point to your router's public IP.

> Okay, so in terminal in haos, it shows a wrong IP? 
>
That should be the IP address of the Nginx docker instance and per the config, it is Nginx that is responsible to serve the pub key.

```
        # static public key for Tesla
        location /.well-known/appspecific/com.tesla.3p.public-key.pem {
            root /share/tesla;
            try_files /com.tesla.3p.public-key.pem =404;
        }
```

I'm a bit out of clue now.... Nginx if it's running should have no issue to serve the key.

**Comment by baylanger** [comment]
@tannerwastaken if from HAOS SSH terminal you run:

```
nc -zv localhost 443
```

Does it return:
`localhost ([::1]:443) open`

I confirm 172.30.32.3 is the IP address of the Nginx add-on.

**Reply by tannerwastaken** [reply]
@baylanger no, it returns the following: 
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/121889392/34ce0bcc-c014-47a4-886f-53f1ed72001f)

**Reply by baylanger** [reply]
What is the full output from the cmd : `docker ps | grep nginx`

**Reply by tannerwastaken** [reply]
Here is output 
<img width="1187" alt="Screenshot 2024-03-30 at 10 31 46 PM" src="https://github.com/llamafilm/tesla-http-proxy-addon/assets/121889392/ee156bf7-e487-4cb5-a927-e54188303c51">

**Reply by baylanger** [reply]
Why is it listening on port 8126? It must be listening on 443. That's probably why things are broken. If you have something else on port 443, either you yank it out or figure out how to configure so that requests for the "Tesla pubkey" goes to Nginx.

**Reply by tannerwastaken** [reply]
Changed it, same results.

**Reply by baylanger** [reply]
Is your output now similar this?
```
XXX  homeassistant/amd64-addon-nginx_proxy:3.9.0 "/init" 9 days ago   Up 9 days 0.0.0.0:443->443/tcp, :::443->443/tcp addon_core_nginx_proxy
```

By default Nginx should have been on port 443, if you made any manual changes, I suggest you delete the Nginx add-on and re-install it. Make sure to use the settings mentioned in DOCS.md.

**Reply by baylanger** [reply]
Not sure what you mean by "Yes." and by " still get the same fix?? ".

1. `docker ps` output shows exactly `0.0.0.0:443->443/tcp, :::443->443/tcp` ?
2. You did make manual changes? If so, can you uninstall/re-install from scratch?

Please elaborate as much.

**Reply by baylanger** [reply]
After moving Nginx to the default 443 port, what's the output of this `docker exec addon_c03d64a7_tesla_http_proxy curl -v https://tesla.thecolorsofyou.com/.well-known/appsp ecific/com.tesla.3p.public-key.pem`

If the above still fails, what's the output in HAOS SSH terminal for `curl -v https://tesla.thecolorsofyou.com/.well-known/appsp ecific/com.tesla.3p.public-key.pem`

If it returns your pub key, then consider to uninstall/reinstall the Tesla HTTP Proxy and afaik it should now work. If the key still isn't returned when running curl, then I'm not sure I can still help.

**Comment by mheloy** [comment]
I've got the same issue wherein "https://tesla.example.com/.well-known/appspecific/com.tesla.3p.public-key.pem" seems not reachable...

I've tried the domain: core-nginx-proxy in NGINX, but when I do this, my HA IOS app saying the cert is not valid

My setup
PROXMOX VM running HAOS
DuckDNS
NGINX
No Nabucasa


DuckDNS
domains:
  - melha.duckdns.org
token: xxxxxxxxxxx
aliases:
  - domain: meltesla.melha.duckdns.org
    alias: melha.duckdns.org
lets_encrypt:
  accept_terms: true
  algo: secp384r1
  certfile: fullchain.pem
  keyfile: privkey.pem
seconds: 300


NGINX
domain: "*.duckdns.org"
hsts: max-age=31536000; includeSubDomains
certfile: fullchain.pem
keyfile: privkey.pem
cloudflare: false
customize:
  active: true
  default: nginx_proxy_default*.conf
  servers: nginx_proxy/*.conf

with this setup, my certs are valid for both melha.duckdns.org and meltesla.melha.duckdns.org as per SSL cert checker
I'm a noob around certs so be gentle... :D
any advise?
